src/coffee-script.coffee

```diff
@@ -156,101 +153,13 @@ exports.nodes = withPrettyErrors (source, options) ->
   else
     parser.parse source
 
-# Compile and execute a string of CoffeeScript (on the server), correctly
-# setting `__filename`, `__dirname`, and relative `require()`.
-exports.run = (code, options = {}) ->
-  mainModule = require.main
+# Compile and execute a string of CoffeeScript 
+exports.run = (code, options = {}) -> throw new Error 'Not implemented'
```

Comment:
The idea is that `coffee-script.coffee` will contain only platform-independent code. Any platform-specific code should be moved elsewhere. Throwing an error in those "abstract" method prevents developers from accidentally require `lib/coffee-script/coffee-script` and not knowing what's wrong.